HealthcareBase/Service/HospitalResourcesService/EquipmentService/EquipmentService.cs

```diff
@@ -69,10 +63,36 @@ private void DeleteFromHospitalizations(EquipmentUnit equipmentUnit)
 
         public void DeleteByType(EquipmentType equipmentType)
         {
-            equipmentType = equipmentTypeRepository.Repository.GetByID(equipmentType.GetKey());
-            foreach (var equipmentUnit in equipmentUnitRepository.Repository.GetAll())
-                if (equipmentUnit.EquipmentType.Equals(equipmentType))
-                    Delete(equipmentUnit);
+           
+        }
+
+        public IEnumerable<EquipmentDto> GetEquipmentByRoomId(int roomId)
+        {
+            return equipmentUnitRepository.Repository.GetColumnsForMatching(
+                condition: equipment => equipment.CurrentLocationId == roomId,
+                selection: equipment => new EquipmentDto()
+                {
+                    Id = equipment.Id,
+                    RoomId = equipment.CurrentLocation.Id,
+                    Name = equipment.EquipmentType.Name,
+                    Quantity = 0
+                }
+            ); ;
+        }
```

Comment:
This method should not be available to everyone. The reason for this is that it does not return data that is correct, but data that needs to be further processed (the amount is always equal to zero). Either you declare this function private or you will calculate the quantity for each element in it